src/wallet/wallet.cpp

```diff
@@ -2740,23 +2715,56 @@ MinedType GetGeneratedType(const uint256& tx, unsigned int vout)
     // Side/Split Stake Support
     else if (wallettx.vout.size() >= 3)
     {
-        // Split Stake
-        if (wallettx.vout[vout].scriptPubKey == wallettx.vout[1].scriptPubKey)
+        // If output 1 is mine and the pubkey (address) for the output is the same as
+        // output 1, it is a split stake return from my stake.
+        if (wallet->IsMine(wallettx.vout[1]) != ISMINE_NO
+                && wallettx.vout[vout].scriptPubKey == wallettx.vout[1].scriptPubKey)
         {
             if (blkindex->nResearchSubsidy == 0)
                 return MinedType::POS;
 
             else
                 return MinedType::POR;
         }
-
         else
         {
-            if (blkindex->nResearchSubsidy == 0)
-                return MinedType::POS_SIDE_STAKE;
+            // If the coinstake return is not mine... (i.e. someone else...)
+            if (wallet->IsMine(wallettx.vout[1]) == ISMINE_NO)
+            {
+                // ... but the output is mine, then this must be a
+                // received sidestake from the staker.
+                if (wallet->IsMine(wallettx.vout[vout]) != ISMINE_NO)
+                {
+                    if (blkindex->nResearchSubsidy == 0)
+                        return MinedType::POS_SIDE_STAKE_RCV;
 
-            else
-                return MinedType::POR_SIDE_STAKE;
+                    else
+                        return MinedType::POR_SIDE_STAKE_RCV;
+                }
+            }
+            // else the coinstake is mine...
+            else if (wallet->IsMine(wallettx.vout[1]) != ISMINE_NO)
+            {
+                // ... but the output is not mine, then this must be a
+                // sidestake sent to someone else.
+                if (wallet->IsMine(wallettx.vout[vout]) == ISMINE_NO)
+                {
+                    if (blkindex->nResearchSubsidy == 0)
+                        return MinedType::POS_SIDE_STAKE_SEND;
+
+                    else
+                        return MinedType::POR_SIDE_STAKE_SEND;
+                }
+                // ... you can also sidestake back to yourself...
+                else if (wallet->IsMine(wallettx.vout[vout]) != ISMINE_NO)
```

Comment:
```suggestion
                else
```
Same here with L2750.